msticpy/data/drivers/sentinel_query_reader.py

```diff
@@ -0,0 +1,452 @@
+
+    Returns
+    -------
+    str
+        Returns the inputted string with non-alphanumeric characters removed and
+        spaces replaced with an underscore
+
+    """
+    del_chars_pattern = r"""[%*\*'\"\\.()[\]{}-]"""
+    repl_str = re.sub(del_chars_pattern, "", qname)
+    repl_str = repl_str.replace(" ", "_").replace("__", "_")
+    return repl_str
+
+
+def _organize_query_list_by_folder(query_list: list) -> dict:
+    """
+    Create a dictionary mapping each folder name with related SentinelQuery objects.
+
+    Parameters
+    ----------
+    query_list : list
+        List of SentinelQuery objects returned by import_sentinel_queries()
+
+    Returns
+    -------
+    dict
+        Returns a dictionary mapping each folder name with the SentinelQuery objects associated
+        with it
+
+    """
+    queries_by_folder = {}
+    for query in query_list:
+        if query.folder_name == "":
+            print(query)
+        if query.folder_name not in queries_by_folder:
+            queries_by_folder[query.folder_name] = [query]
+        else:
+            queries_by_folder[query.folder_name].append(query)
+
+    return queries_by_folder
+
+
+def _create_queryfile_metadata(folder_name: str) -> dict:  # type: ignore
+    """
+    Generate metadata section of the YAML file for the given folder_name.
+
+    Parameters
+    ----------
+    folder_name : str
+        Either "Hunting Queries" or "Detections" or otherwise named query category
+
+    Returns
+    -------
+    dict
+        Returns a generated metadata section for the YAML files in the given folder_name
+
+    """
+    dict_to_write = {"metadata": {}, "defaults": {}, "sources": {}}  # type: ignore
+    dict_to_write["metadata"]["version"] = 1  # write update version functionality
+    dict_to_write["metadata"]["description"] = "Sentinel Alert Queries - " + folder_name
+    dict_to_write["metadata"]["data_environments"] = [
+        "MSSentinel"
+    ]  # write how to get this and what type this should be
+    dict_to_write["metadata"]["data_families"] = folder_name
+    dict_to_write["metadata"]["last_updated"] = str(datetime.now())
+    return dict_to_write
+
+
+def write_to_yaml(query_list: list, query_type: str, output_folder: str) -> bool:
+    """
+    Write out generated YAML files of the given query_list into the given output_folder.
+
+    Parameters
+    ----------
+    query_list : list
+        List of SentinelQuery attr objects generated by import_sentinel_queries()
+    query_type : str
+        Either "Hunting Queries" or "Detections" or otherwise named query category
+    output_folder : str
+        The name of the folder you want the written YAML files to be stored in
+
+    Returns
+    -------
+    bool
+        True if succeeded; False if an error occurred
+
+    """
+    query_dict_by_folder = _organize_query_list_by_folder(query_list)
+    all_folders = query_dict_by_folder.keys()
+
+    for source_folder in all_folders:
```

Comment:
Might be good to have tqdm on this loop since I would guess it's quite time-consuming.

Reply:
For some reason writing out the files actually takes < 2 seconds, but I'll log out the folders being written instead of printing so that the general function should be able to see those too.